File: pset6/readability/readability.py
from cs50 import get_string

# Get text from user
text = get_string("Text: ")
# Calculate number of letters, words and sentences
letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if text[i].isalpha():
        letters += 1

    if i < (len(text) - 1):
        if text[i] == " " and (text[i + 1] != " "):
            words += 1

    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

# Calculate the index
index = 0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {int(round(index))}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    //predefine number of letters, words and sentences
    int letters = 0, words = 1, sentences = 0;

    //prompt user for text
    string text = get_string("Text: ");

    //calculate number of letters, words and sentences
    for (int i = 0; i < strlen(text); i++)
    {
        //check number of alphabetic letters
        if (isalpha(text[i]))
        {
            letters++;
        }

        //check if there is a non-space character after space, if yes then increase word count by 1
        if (isspace(text[i]) && ~isspace(text[i + 1]))
        {
            words++;
        }

        //check for period, exclamation point, question mark, if detected then increase sentence count by 1
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    //printf("%i letter(s)\n", letters);
    //printf("%i word(s)\n", words);
    //printf("%i sentence(s)\n", sentences);

    //calculate Coleman-Liau index
    float index = 0.0588 * (float) letters / (float) words * 100 - 0.296 * (float) sentences / (float) words * 100 - 15.8;

    //Check for reading level then print appropriate result
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}